wcomponents-core/src/main/java/com/github/bordertech/wcomponents/render/webxml/WTableRenderer.java

```diff
@@ -439,6 +439,14 @@ private void paintColumnHeaderFooter(final WTable table, final WebXmlRenderConte
 			// Footers
 			xml.appendTag("tfoot");
 			xml.appendTag("tr");
+			// Allow for expandable column
+			if (table.getExpandMode() != ExpandMode.NONE) {
+				xml.append("<td/>");
+			}
+			// Allow for selectable column
+			if (table.getSelectMode() != SelectMode.NONE) {
+				xml.append("<td/>");
```

Comment:
So the problem with rendering a HTML tfoot and associated elements is they have to be in the correct form and the td element is not self-closing so should be output as `<td></td>` rather than `<td/>`. It will work with the self-closing element but it isn't right.

Reply:
Agree. Will fix. What a mess self closing tags are HTML vs XHTML.